### PR TITLE
Add tooltip to help icon

### LINKED ISSUE #3573: Add clarifying tooltips to header navigation
**Is your feature request related to a problem? Please describe.**

Related to https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3557 , proper tool tips do not appear on hover for Menu or Help icons. 

**Describe the solution you'd like**

Similar to the tool tip that appears on "Home" icon in the stacked nav or on the OpenSearch logo icon in the single nav, add tool tips that describe, in the simplest terms, what the icons represent. 

**Describe alternatives you've considered**

In order to further disambiguate the purpose of these icons, the UX team is currently working on redesigning our icon offering. 
Example: https://github.com/opensearch-project/ux/issues/11

However, regardless of the direction that an icon redesign may take, proper tool tips are a must-have. 

**Additional context**
Screen Shot 2023-03-09 at 10 40 00 AM

### PR DESCRIPTION
**Pull request by sayuree**
### Description
Added tooltip to a Help icon
 
### Issues Resolved
fixes #3573 
 
### Check List
- [ ] All tests pass
  - [x] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/header_help_menu.tsx
Comment (ashwin-pc) [inline src/core/public/chrome/ui/header/header_help_menu.tsx:331]:
```suggestion
        <EuiIcon type="help" size="m" title={intl.formatMessage({
          id: 'core.ui.chrome.headerGlobalNav.helpMenuButtonTitle',
          defaultMessage: 'Help menu',
        })} />
```

**Review by KrooshalUX — CHANGES_REQUESTED** [review]
Please see comment in main issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3573#issuecomment-1474408957

**Comment by abbyhu2000** [comment]
Hi @sayuree, will you be able to make the suggested changes for this PR?

**Comment by sayuree** [comment]
@abbyhu2000 yes, I am able to finish it.

File: src/core/public/chrome/ui/header/header_help_menu.tsx
Comment (joshuarrrr) [inline src/core/public/chrome/ui/header/header_help_menu.tsx:336]:
```suggestion
            defaultMessage: 'Help',
```

I think the agreed upon text is simply "Help" - @KrooshalUX can you confirm?

Comment (KrooshalUX) [reply src/core/public/chrome/ui/header/header_help_menu.tsx]: It should be `Help`, yes.

File: src/core/public/chrome/ui/header/header.tsx
Comment (abbyhu2000) [inline src/core/public/chrome/ui/header/header.tsx:177]:
```suggestion
                    title={i18n.translate('core.ui.primaryNav.menu', {
```

Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/header.tsx:177]: I don't think this should be `core.ui.primaryNav.toggleNavAriaLabel`

**Review by abbyhu2000 — CHANGES_REQUESTED** [review]

**Comment by abbyhu2000** [comment]
@sayuree Can you also add UI screenshots to show the tooltips on both the help icon and menu icon?

**Comment by sayuree** [comment]
> @sayuree Can you also add UI screenshots to show the tooltips on both the help icon and menu icon? Thank you!

Here are the screenshots:
Screenshot from 2023-03-28 15-31-40
Screenshot from 2023-03-28 15-31-48

**Comment by joshuarrrr** [comment]
@KrooshalUX Screenshots provided.

**Review by KrooshalUX — DISMISSED** [review]
LGTM

**Comment by abbyhu2000** [comment]
LGTM, will approve after the tests finish running

**Comment by joshuarrrr** [comment]
@sayuree There are some snapshot tests that need to be updated: https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4532314219/jobs/8036242750?pr=3626

You can update snapshots via `yarn run test:jest -u`

File: src/dev/build/tasks/os_packages/create_os_package_tasks.test.ts
Comment (joshuarrrr) [inline src/dev/build/tasks/os_packages/create_os_package_tasks.test.ts:124]: @sayuree This snapshot file change should be removed, as its unrelated to the change.

File: src/plugins/vis_type_tagcloud/public/components/__snapshots__/tag_cloud_visualization.test.js.snap
Comment (joshuarrrr) [inline src/plugins/vis_type_tagcloud/public/components/__snapshots__/tag_cloud_visualization.test.js.snap:7]: This snapshot file change should be removed, as its unrelated to the change.

File: src/setup_node_env/__snapshots__/node_version_validator.test.js.snap
Comment (joshuarrrr) [inline src/setup_node_env/__snapshots__/node_version_validator.test.js.snap:141]: This snapshot change should be removed, as its unrelated to the change.

**Review by joshuarrrr — COMMENTED** [review]
@sayuree When you updated the snapshots, some irrelevant changes are now in the PR. Those file changes should be removed from the PR.

**Comment by sayuree** [comment]
@joshuarrrr @abbyhu2000 I have rebased the PR, removed unnecessarily updated snapshots. Tests should approve now.

**Review by abbyhu2000 — APPROVED** [review]

**Review by ananzh — APPROVED** [review]
LGTM

**Pull request merged**
